contracts/OddsLib.sol

```diff
@@ -53,4 +53,32 @@ library OddsLib {
         // Return the odds need to generate this adjusted payout with the given wager
         return Math.max(1 * PRECISION, (adjustedPayout * PRECISION) / wager);
     }
+
+    // Assuming that the market previously had targetMargin, then after some runners have been scratched, correct the odds of this runner to make up the margin again
+    function rebaseOddsWithScratch(uint256 odds, uint256 scratchedOdds, uint256 targetMargin) external pure returns (uint256) {
+        uint256 newMargin = targetMargin - PRECISION / scratchedOdds;
+        return changeMargin(odds, newMargin, targetMargin);
+    } 
+
+    function changeMargin(uint256 odds, uint256 margin, uint256 targetMargin) public pure returns (uint256) {
+        return addMargin(removeMargin(odds, margin), targetMargin);
+    }
+
+    function removeMargin(uint256 odds, uint256 margin) public pure returns (uint256) {
+        return odds * margin / PRECISION;
+    }
+
+    function addMargin(uint256 odds, uint256 margin) public pure returns (uint256) {
+        return odds * PRECISION / margin;
+    }
+
+    // Given an array of odds, return the margin
+    function getMargin(uint256[] calldata odds) public pure returns (uint256) {
+        uint256 total = 0;
+        for (uint256 i = 0; i < odds.length; i++) {
```

Comment:
Its better to "cache" this value, as reading the .length property uses more gas.

```suggestion
        uint256 count = odds.length;
        for (uint256 i = 0; i < count; i++) {
```